### PR TITLE
migrate zypprepo from type only to type and provider

### LINKED ISSUE #5: ability to remove repo
We need an ensure => present/absent in order to be able to remove a zypprepo

### LINKED ISSUE #9: option to remove all repositories that are not managed by puppet

### PR DESCRIPTION
**Pull request by tuxmea**
#### Pull Request (PR) description

adopt the yumrepo type/provider to zypprepo.

Open issue: repo files will be emptied, but not removed when using
ensure => absent

#### This Pull Request (PR) fixes the following issues
Fixes #5
Fixes #9

### CODE REVIEW

File: lib/puppet/provider/zypprepo/inifile.rb [hunk 1]
```diff
@@ -0,0 +1,312 @@
+# Description of zypper repositories
+require 'pry'
```
Comment (bastelfreak) [inline lib/puppet/provider/zypprepo/inifile.rb:2]: is that a debugging-leftover?

**Comment by bastelfreak** [comment]
@alexjfisher can you take a look? You're my type and provider expert!

File: lib/puppet/provider/zypprepo/inifile.rb (hunk 1 above)
Comment (tuxmea) [reply lib/puppet/provider/zypprepo/inifile.rb]: yes. fixed.

**Comment by tuxmea** [comment]
Note: resource collections are working properly (puppet resource zypprepo).
Empty files will be ignored.

**Comment by alexjfisher** [comment]
> @alexjfisher can you take a look? You're my type and provider expert!

You're *too* kind.  No, really! :)

I can't see any issues, but any chance of a basic acceptance test? (Not that I'm sure we've ever done any for SUSE!)

**Review by alexjfisher — APPROVED** [review]
Tests would be nice, but I understand if you're short on time, so I'm ok approving this without.
I'm not in a position to test this, but the great documentation shows you've put a lot of thought into it.  Thanks!

**Comment by tuxmea** [comment]
Short on time and urgently needed for a customer.
I will add tests definitely this year.

**Comment by alexjfisher** [comment]
> I will add tests definitely this year.

Excellent!  Thanks!  I'll see about that release.

File: lib/puppet/type/zypprepo.rb [hunk 2]
```diff
@@ -1,298 +1,103 @@
...
+    newvalue(%r{.*}) {}
+  end
 
-    newproperty(:repo_gpgcheck, parent: Puppet::IniProperty) do
-      desc "Whether to check the GPG signature on the repository
-         metadata, as represented by a `0` or `1`.
-        #{ABSENT_DOC}"
-      newvalue(:absent) { self.should = :absent }
-      newvalue(%r{(0|1)}) {}
-    end
```
Comment (mx-psi) [inline lib/puppet/type/zypprepo.rb:253]: @tuxmea This reverted #48. Was it an overlook or is there any reason why it was reverted?

Comment (tuxmea) [reply lib/puppet/type/zypprepo.rb:253]: seems as if I overlooked two properties. going to update with the actual open PR

Comment (mx-psi) [reply lib/puppet/type/zypprepo.rb:253]: 🙇 Thanks!

Comment (tuxmea) [reply lib/puppet/type/zypprepo.rb:253]: readded properties to #55 
At the moment I only have one more issue: correcting e.g. priority is not working.

**Pull request merged**